### PR TITLE
Fix #3448 - @northwestwitch patch

### LINKED ISSUE #3448: Missing genes in PanelApp panel
Hello, me again!

I'm experiencing some missing gene's in the [PanelApp](https://panelapp.genomicsengland.co.uk/) panels loaded into scout. I was testing with scout [v4.54](https://github.com/Clinical-Genomics/scout/tree/v4.54) and tried again today with your most recent release [v4.56](https://github.com/Clinical-Genomics/scout/tree/v4.56) (and re-loading the PanelApp panels with `scout -db load panel --panel-app`) and still seem to be experiencing the same thing. Apologies I'm not giving a super tidy reproducible example, just explaining what I've observed. If it's not easily reproduced I'll spend more time creating a reproducible example :)

I checked several panels in the scout user interface, (they're all the most recent versions) here are my observations.

PanelApp panels in scout with all the genes that are found in the green/high confidence list in the PanelApp website:

- [Arrhythmogenic cardiomyopathy](https://panelapp.genomicsengland.co.uk/panels/134/)
- [Atypical haemolytic uraemic syndrome](https://panelapp.genomicsengland.co.uk/panels/139/)
- [Arthrogryposis](https://panelapp.genomicsengland.co.uk/panels/258/)
- [Familial hidradenitis suppurativa](https://panelapp.genomicsengland.co.uk/panels/23/)
- [GI tract tumours](https://panelapp.genomicsengland.co.uk/panels/254/)
- [Infantile nystagmus](https://panelapp.genomicsengland.co.uk/panels/246/)
- [Inherited bleeding disorders](https://panelapp.genomicsengland.co.uk/panels/175/)
- [Inherited ovarian cancer (without breast cancer)](https://panelapp.genomicsengland.co.uk/panels/143/)
- [Intracerebral calcification disorders](https://panelapp.genomicsengland.co.uk/panels/315/)
- [Limb girdle muscular dystrophy](https://panelapp.genomicsengland.co.uk/panels/185/)

PanelApp panels in scout that are missing genes that are found in the green/high confidence list in the PanelApp website:

- [CAKUT](https://panelapp.genomicsengland.co.uk/panels/234/)
- [Clefting](https://panelapp.genomicsengland.co.uk/panels/81/)
- [Brain channelopathy](https://panelapp.genomicsengland.co.uk/panels/90/)
- [Mitochondrial disorders](https://panelapp.genomicsengland.co.uk/panels/112/)

Genes that that were missing from these panels in scout that were "CNV Loss":

- [17q12 recurrent (RCAD syndrome) region (includes HNF1B) Loss](https://panelapp.genomicsengland.co.uk/panels/234/region/ISCA-37432-Loss/) (I'm assuming the HGNC symbol is [HNF1B](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:11630))
- [22q11.2 recurrent (DGS/VCFS) region (proximal, A-B) (includes TBX1) Loss](https://panelapp.genomicsengland.co.uk/panels/81/region/ISCA-37433-Loss/) (I'm assuming the HGNC symbol is [TBX1](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:11592))
- [22q11.2 recurrent (DGS/VCFS) region (proximal, A-D) (includes TBX1) Loss](https://panelapp.genomicsengland.co.uk/panels/81/region/ISCA-37446-Loss/) ( I'm assuming the HGNC symbol is [TBX1](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:11592))
- [Xp11.23 region (includes MAOA and MAOB) Loss](https://panelapp.genomicsengland.co.uk/panels/90/region/ISCA-37468-Loss/) (I'm assuming the HGNC symbol is [MAOA](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:13243) and [MAOB](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:6834))
- [ISCA-37440-Loss 2p21 region (includes PREPL and SLC3A1) Loss](https://panelapp.genomicsengland.co.uk/panels/112/region/ISCA-37440-Loss/) (I'm assuming the HGNC symbol is [PREPL](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:30228) and [SLC3A1](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:11025))

Genes that that were missing from these panels in scout that were  "CNV Gain":

- [22q11.21 recurrent (Cat eye syndrome) region (includes CECR2) Gain](https://panelapp.genomicsengland.co.uk/panels/81/region/ISCA-37393-Gain/) (I'm assuming the HGNC symbol is [CECR2](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:1840))
- [8p23.1 recurrent region (includes GATA4) Gain](https://panelapp.genomicsengland.co.uk/panels/81/region/ISCA-37423-Gain/) (I'm assuming the HGNC symbol is [GATA4](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:4173))
- [7q36.3 ZRS (SHH cis-regulatory) duplication region (within LMBR1 intron 5) Gain](https://panelapp.genomicsengland.co.uk/panels/81/region/ISCA-37467-Gain/) (I'm assuming the HGNC symbol is [LMBR1](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:13243))

Genes that that were missing from these panels that were  "STR":

- [ATN1_CAG](https://panelapp.genomicsengland.co.uk/panels/90/str/ATN1_CAG/) (ATN1 is [in OMIM](https://omim.org/entry/607462) and has an [associated HGNC gene symbol (ATN1)](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:3033))
- [CACNA1A_CAG](https://panelapp.genomicsengland.co.uk/panels/90/str/CACNA1A_CAG/) (CACNA1A is [in OMIM](https://omim.org/entry/601011) and has an [associated HGNC gene symbol (CACNA1A)](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:1388))
- [CSTB_CCCCGCCCCGCG](https://panelapp.genomicsengland.co.uk/panels/90/str/CSTB_CCCCGCCCCGCG/) (CSTB is [in OMIM](https://omim.org/entry/601145) and has an [associated HGNC gene symbol (CCSTB)](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:2482))
- [TBP_CAG](https://panelapp.genomicsengland.co.uk/panels/90/str/TBP_CAG/) (TBP is [in OMIM](https://omim.org/entry/600075) and has an [associated HGNC gene symbol (TBP)](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:11588))

Genes that that were missing from these panels that were other types:

- [C5orf42](https://panelapp.genomicsengland.co.uk/panels/81/gene/C5orf42/) (C5orf42 is [in OMIM](https://omim.org/entry/614571) and has an [associated HGNC gene symbol (CPLANE1)](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:25801))
- [C12orf65](https://panelapp.genomicsengland.co.uk/panels/112/gene/C12orf65/) (C12orf65 is [in OMIM](https://www.omim.org/entry/613541?search=c12orf65&highlight=c12orf65) and has an [associated HGNC gene symbol (MTRFR)](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:26784))
- [C19orf70](https://panelapp.genomicsengland.co.uk/panels/112/gene/C19orf70/) (C19orf70 is [in OMIM](https://www.omim.org/entry/616658?search=c19orf70&highlight=c19orf70) and has an [associated HGNC gene symbol (MICOS13)](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:33702))

22q11.21 recurrent (Cat eye syndrome) region (includes CECR2) Gain, 8p23.1 recurrent region (includes GATA4) Gain and 7q36.3 ZRS (SHH cis-regulatory) duplication region (within LMBR1 intron 5) Gain are present in scout and can be accessed by what I assume are their HGNC symbols. For example:

```bash
scout --port 27018 -db test-database view hgnc --hgnc-symbol CECR2
```

My partial output:

```bash
#hgnc_id        hgnc_symbol     aliases
1840    CECR2   CECR2, KIAA1740
```

```bash
scout --port 27018 -db test-database view hgnc --hgnc-id 1840
```

My partial output:

```bash
#hgnc_id        hgnc_symbol     aliases
1840    CECR2   CECR2, KIAA1740
```

ATN1_CAG, CACNA1A_CAG, CSTB_CCCCGCCCCGCG and TBP_CAG are present in scout and can be accessed by what I assume are their HGNC symbols if you drop everything after the underscore. For example:

```bash
scout --port 27018 -db test-database view hgnc --hgnc-symbol ATN1_CAG
```

My partial output:

```bash
#hgnc_id        hgnc_symbol     aliases
2022-06-16 17:01:03 leviathan scout.commands.view.hgnc[2619765] INFO No results found
```

```bash
scout --port 27018 -db test-database view hgnc --hgnc-id 3033
```

My partial output:

```bash
#hgnc_id        hgnc_symbol     aliases
3033    ATN1    B37, ATN1, D12S755E, DRPLA
```

```bash
scout --port 27018 -db test-database view hgnc --hgnc-symbol ATN1
```

My partial output:

```bash
#hgnc_id        hgnc_symbol     aliases
3033    ATN1    B37, ATN1, D12S755E, DRPLA
```

C5orf42, C12orf65 and C19orf70 are present in scout and can be accessed by the HGNC symbols used by PanelApp, their current HGNC symbols and their HGNC id. For example:

```bash
scout --port 27018 -db test-database view hgnc --hgnc-symbol C5orf42
```

My partial output:

```bash
#hgnc_id        hgnc_symbol     aliases
25801   CPLANE1 JBTS17, CPLANE1, FLJ13231, Hug, C5orf42
```

```bash
scout --port 27018 -db test-database view hgnc --hgnc-symbol CPLANE1
```

My partial output:

```bash
#hgnc_id        hgnc_symbol     aliases
25801   CPLANE1 JBTS17, Hug, CPLANE1, C5orf42, FLJ13231
```

```bash
scout --port 27018 -db test-database view hgnc --hgnc-id 25801
```

My partial output:

```bash
#hgnc_id        hgnc_symbol     aliases
25801   CPLANE1 JBTS17, Hug, CPLANE1, C5orf42, FLJ13231
```

It looks like this is [the function in scout responsible for loading the PanelApp panels](https://github.com/Clinical-Genomics/scout/blob/main/scout/load/panel.py#L108) and [where this function is called](https://github.com/Clinical-Genomics/scout/blob/c89cc3c0025ea918d1430dec07ccbcd92326ca0d/scout/commands/load/panel.py#L69). My python isn't great so I haven't looked much further into the code. Either way it seems like there are some "edge cases" that this function isn't currently accounting for so they're just getting considered not present in the database and skipped when loading the PanelApp panels. Another thing to note, I didn't see any of these "edge case" genes in the [reduced HGNC dataset that scout uses for the demo database](https://github.com/Clinical-Genomics/scout/blob/main/scout/demo/resources/hgnc_reduced_set.txt) so these situations might be missed if testing with this reduced dataset.

Hope this is useful, again let me know if you have trouble reproducing this and I'll spend some time creating a reproducible example :)

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the paxa software installed on `hasta`:
 - Log in into hasta: `ssh <USER.NAME>@hasta.scilifelab.se`
 - Activate the staging environment with the command `us` (Use Stage)
 - Run the command `paxa` and follow the instructions. Make sure to specify scout-stage as the resource you request and the server cg-vm1 as server.
1. Log out from the hasta server.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, log out from `cg-vm1` and log in again in the `hasta` server, repeat the `hasta` and `paxa` procedure, which will release the allocated resource (scout-stage) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by northwestwitch** [comment]
Unfortunately this fixes the PanelApp error, but creates another in the normal gene loading:

```
2022-06-21 14:03:14 ChiaraRMBP.local scout.utils.link[26460] INFO Add omim info
2022-06-21 14:03:14 ChiaraRMBP.local scout.parse.omim[26460] INFO Get the mim genes
2022-06-21 14:03:14 ChiaraRMBP.local scout.parse.omim[26460] INFO Parsing mim2gene
2022-06-21 14:03:14 ChiaraRMBP.local scout.parse.omim[26460] INFO Number of genes without hgnc symbol 123
2022-06-21 14:03:14 ChiaraRMBP.local scout.parse.omim[26460] INFO Parsing the omim genemap2
Traceback (most recent call last):
  File "/Users/chiararasi/miniconda3/envs/py38/bin/scout", line 33, in <module>
    sys.exit(load_entry_point('scout-browser', 'console_scripts', 'scout')())
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/flask/cli.py", line 579, in main
    return super().main(*args, **kwargs)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/flask/cli.py", line 427, in decorator
    return __ctx.invoke(f, *args, **kwargs)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/commands/update/genes.py", line 161, in genes
    hgnc_genes = load_hgnc_genes(
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/load/hgnc_gene.py", line 75, in load_hgnc_genes
    genes = link_genes(
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/utils/link.py", line 219, in link_genes
    add_omim_info(genes, symbol_to_id, genemap_lines, mim2gene_lines)
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/utils/link.py", line 125, in add_omim_info
    for hgnc_id in get_correct_ids(hgnc_symbol, alias_genes):
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/utils/link.py", line 161, in get_correct_ids
    hgnc_id_info = alias_genes[hgnc_symbol]
KeyError: 'C15orf48'
(py38) chiararasi@ChiaraRMBP:~/Documents/work/GITs/scout$
```

File: scout/utils/link.py [hunk 1]
```diff
@@ -156,8 +156,8 @@ def get_correct_ids(hgnc_symbol, alias_genes):
         hgnc_ids(iterable(int)): Hopefully only one but a symbol could map to several ids
     """
     hgnc_ids = set()
-    hgnc_symbol = hgnc_symbol.upper()
-    if hgnc_symbol in alias_genes:
+    # hgnc_symbol = hgnc_symbol.upper()
+    if hgnc_symbol in alias_genes or hgnc_symbol.upper() in alias_genes:
```
Comment (northwestwitch) [inline scout/utils/link.py:160]: To have a proper fix I think this should be:

```
if hgnc_symbol in alias_genes:
   hgnc_id_info = alias_genes[hgnc_symbol]
if hgnc_symbol.upper() in alias_genes:
   hgnc_id_info = alias_genes[hgnc_symbol.upper()]
```

Comment (dnil) [reply scout/utils/link.py:160]: But also we must still be getting non-upper entries into the alias_genes map. I'll try to take a look whenever I get a moment..

Comment (dnil) [reply scout/utils/link.py:160]: This link side, with OMIM, incomplete penetrance and exac info, seems good on its own, operating with upper case symbols for alias lookup. There is a complementary build aware system for nearly the same in the GeneHandler adapter class. I will try to set this up so we can use it directly instead for the alias collection as well, as the author apparently decided to use it for generating the alias map.

Comment (dnil) [reply scout/utils/link.py:160]: Meh, I give up; too tired. Im afraid I will introduce bugs wrt build version this way. I used a version of your suggestion for now, and we can reconsider tomorrow. Im not sure it is completely build safe still either though, but follows a course of safe improvement. Let's test with like a panel with a hg38 only gene.

File: scout/utils/link.py [hunk 2]
```diff
@@ -148,20 +146,27 @@ def get_correct_ids(hgnc_symbol, alias_genes):
...
     """
     hgnc_ids = set()
-    hgnc_symbol = hgnc_symbol.upper()
+    hgnc_id_info = {}
+
     if hgnc_symbol in alias_genes:
         hgnc_id_info = alias_genes[hgnc_symbol]
-        if hgnc_id_info["true"]:
-            return set([hgnc_id_info["true"]])
+    elif hgnc_symbol.upper() in alias_genes:
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/utils/link.py:161]: I know that I've suggested that patch in the first place, but I regretted it 😆

**Pull request closed without merging**